clusterloader2/pkg/util/parallel.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+// GoroutineGroup executes routines.
+type GoroutineGroup interface {
+	Execute(routine func())
+}
+
+type limitedParallelGroup struct {
```

Comment:
This was introduced due to problem with pod-stores. When `WaitForControlledPodsRunning` is used during scaling RCs, it creates pod-stores for each existing RC and some of the pod-store fail. I though that putting limit on number of running goroutines, will stagger the pod-stores creations.

In original tests this problem didn't exist, cause after scaling one RC single pod-store was created. Scaling RCs was spread over time.

Reply:
I'm not sure why do you say it's a problem now?

Since we are saying that we're only care about changes of a Spec, then change in a Spec of some RC should trigger starting a PodStore.
So spreading those scales over time makes it equivalent to what we have currently.

Is there anything I'm missing?

Reply:
The measurement I implemented works like this:
1. Lists all RCs that satisfy the requirements
2. Starts to verify statuses of each listed RC
3. Gets events for RCs that satisfy the requirements.
- Create runs verifying for given RC
- Update stops verifying for older version (if any is running) and starts new verification
- Delete removes RC info from the list.

When "gather" called, it waits until there is specific number of RCs with all pods running.

The problem is with point 2. This synchronize inner RC list with existing RCs. E.g. if there is 10000 RC running, 10000 pod-stors will be created (at the same time) to verify RC statuses.

Reply:
Having this parallel group defeats the whole purpose of timeout.

In the current implementation of our tests, it won't hurt us, because we start the measurement when we don't have any RC.
If we want to do that correctly, the solution is to share a single pod store (the TODo that I suggested couple days ago: https://github.com/kubernetes/perf-tests/pull/264/files#r227685929), not introducing a limitted parallel group.

Reply:
After offline discussion, we came to agreement  to remove parallel group.